ui/dashboards/src/components/Variables/BuiltinVariableAccordions.tsx

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { BuiltinVariableDefinition } from '@perses-dev/core';
+import {
+  Accordion,
+  AccordionDetails,
+  AccordionSummary,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Typography,
+} from '@mui/material';
+import ExpandMoreIcon from 'mdi-material-ui/ChevronUp';
+import { InfoTooltip } from '@perses-dev/components';
+import { useMemo } from 'react';
+
+type BuiltinVariableAccordionsProps = {
+  builtinVariableDefinitions: BuiltinVariableDefinition[];
+};
+
+export function BuiltinVariableAccordions({ builtinVariableDefinitions }: BuiltinVariableAccordionsProps) {
+  const builtinVariablesBySource = useMemo(() => {
+    const result: Record<string, BuiltinVariableDefinition[]> = {};
+    for (const definition of builtinVariableDefinitions) {
+      const value = result[definition.spec.source];
+      if (value) {
```

Comment:
Nothing special but I guess it might reduce the cyclomatic complexity a little.

```suggestion
      if (value) {
          value.push(definition);
          continue;
      }
      result[definition.spec.source] = [definition];
```